AdobeStockClient/Model/Client.php

```diff
@@ -216,17 +220,24 @@ private function getSearchRequest(SearchCriteriaInterface $searchCriteria): Sear
     }
 
     /**
-     * Retrive array of columns to be requested
+     * Retrieve array of columns to be requested
      *
      * @return array
+     * @throws IntegrationException
      */
     private function getResultColumns(): array
     {
         $resultsColumns = Constants::getResultColumns();
         $resultColumnArray = [];
-        foreach ($this->config->getSearchResultFields() as $field) {
-            $resultColumnArray[] = $resultsColumns[$field];
+        try {
+            foreach ($this->config->getSearchResultFields() as $field) {
+                $resultColumnArray[] = $resultsColumns[$field];
+            }
+        } catch (Exception $exception) {
+            $message = __('An error occurred during result fields matching: %1', $exception->getMessage());
```

Comment:
It would be better to log error stating that "Cannot retrieve the field %field%. It's not available in Adobe Stock SDK." and skip this filed instead of terminating the whole process